### PR TITLE
Cleanup warnings

### PR DESCRIPTION
**Pull request by gliptak**

### CODE REVIEW

**Comment by sns-seb** [comment]
Hello,

Thank you for this PR, fixing all these warnings in old code is a "simple" yet important contribution.

But, as is, I can not review it.

Could you please have a look at the failing continuous integration build on your Pull Request, code does not compile in its current state.

File: server/sonar-server/src/main/java/org/sonar/server/db/BaseDao.java [hunk 1]
```diff
@@ -215,7 +215,7 @@ private void update(DbSession session, DTO item, Date now) {
       item.setUpdatedAt(now);
       doUpdate(session, item);
       if (hasIndex()) {
-        session.enqueue(new UpsertDto(getIndexType(), item));
+        session.enqueue(new UpsertDto<DTO>(getIndexType(), item));
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/db/BaseDao.java]: For your information BaseDao is being dropped. No need to make effort to improve it.

**Comment by gliptak** [comment]
I will work on correcting Travis build. Thanks

**Comment by gliptak** [comment]
Hmm, my local build doesn't show these errors ...

**Comment by gliptak** [comment]
One of the build  fails with:

[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:0.0.20:npm (npm install) on project sonar-web: Failed to run task: 'npm install --color=false' failed. (error code 1) -> [Help 1]

**Comment by sns-seb** [comment]
Hi @gliptak,

Indeed, the error you report in your last comment has nothing to do with your PR. This is a little instability issue we are about to fix. Still, the compilation did fail when I made my comment.

I see the build is now ok, I'll will review your PR and let you know.

**Comment by gliptak** [comment]
Success in one configuration ... Maybe the unit tests are still unstable?

**Comment by sns-seb** [comment]
Hi @gliptak ,

You need to rebase on master to fix these build errors. There was an issue with the Travis build configuration which was fixed 4 or 5 hours ago.

**Comment by gliptak** [comment]
Rebased.

**Comment by sns-seb** [comment]
merged, thank you for your contribution @gliptak

**Pull request merged**